wordpress-mobile/WordPress-iOS
# Add authentication support to image downloads for Atomic Private Sites.

Closes #13571 

This PR forces authentication for all our media downloads from WP sites.  In particular, this ensures we can make media requests to Atomic Private sites either directly or through Photon URLs (which are replaced by a proxy URL since Photon currently does not support Atomic Private Sites).

## Additional Info:

I had to change some calls to download images from `UIImageView+Networking.swift` to AlamofireImage.  The reason was that I was seeing a lot of issues that were NOT introduced by my code, but that were made more noticeable by it.

Please DM if you want to know the details, as the conversation around this happened in private.

As a side note, I'm not sure if it's just me but images are loading extremely fast after the switch, and the Apps feels a lot lighter and snappier.

## To test:

Try to make sure you test with a public site, a private site, an atomic private site, and a self-hosted site.

### Test 1:

1. Create a post with images from different sources (device media, WP library, etc).
2. Publish it.
3. Make sure the image looks fine.

### Test 2:

Open the media library for your  site and make sure it looks good.

### Test 3:

Set a site icon for your site and make sure it loads fine.

## PR submission checklist:
